src/librustdoc/visit_ast.rs

```diff
@@ -103,15 +99,15 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
 
     fn visit_mod_contents(
         &mut self,
-        vis: &hir::Visibility<'_>,
         id: hir::HirId,
         m: &'tcx hir::Mod<'tcx>,
         name: Symbol,
     ) -> Module<'tcx> {
         let mut om = Module::new(name, id, m.inner);
+        let def_id = id.expect_owner();
```

Comment:
I'm afraid I'm not really familiar with the method you're referring to? I couldn't find anything like that, but that doesn't mean much, given how much of a maze `TyCtxt` is. In any case, wouldn't it be better to check the invariant that the module is its own owner rather than assuming it?

Reply:
Ah, it's `tcx.hir().local_def_id_to_hir_id()`.

> In any case, wouldn't it be better to check the invariant that the module is its own owner rather than assuming it?

If modules aren't their own owners, then basically all of rustdoc's code is broken and every test will fail. I don't think we need to assert it for every module rustdoc ever encounters.

Reply:
Alright!